### PR TITLE
fix: pay now button validations

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
Previously pay now button don't have any validations and used to get clear the SDK without doing anything. 

Added checks for the following button - 

Screenshot 2024-05-31 at 12 10 05 PM

### CODE REVIEW

File: src/screens/SDKPayment/WebSDK.res [hunk 1]
```diff
@@ -161,28 +161,28 @@ module CheckoutForm = {
...
+          | Some(str) =>
+            switch str {
+            | "failed" => setPaymentStatus(_ => FAILED("Failed"))
+            | "succeeded" => setPaymentStatus(_ => SUCCESS)
+            | _ => setPaymentStatus(_ => CUSTOMSTATE)
+            }
+          | None => setPaymentStatus(_ => CUSTOMSTATE)
           }
-        | None => setPaymentStatus(_ => CUSTOMSTATE)
+          setClientSecret(_ => None)
```
Comment (JeevaRamu0104) [inline src/screens/SDKPayment/WebSDK.res:185]: why its moved over here

Comment (PritishBudhiraja) [reply src/screens/SDKPayment/WebSDK.res:185]: Because in case of validation error it was getting set to null. We don't need to set to null.

Comment (JeevaRamu0104) [reply src/screens/SDKPayment/WebSDK.res:185]: at any point we need to regenerate new client secret if we get some error

Comment (PritishBudhiraja) [reply src/screens/SDKPayment/WebSDK.res:185]: yes if
`if errorDict->getOptionString("type") !== Some("validation_error") {`

This comes just don't redirect or clear the client secret.

Comment (JeevaRamu0104) [reply src/screens/SDKPayment/WebSDK.res:185]: can you add some description or screenshot

Comment (PritishBudhiraja) [reply src/screens/SDKPayment/WebSDK.res:185]: Updated

**Pull request merged**